src/Microsoft.Owin.StaticFiles/ICustomEtagProvider.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved. See License.txt in the project root for license information.
+
+using Microsoft.Owin.FileSystems;
+
+namespace Microsoft.Owin.StaticFiles
+{
+    /// <summary>
+    /// Custom provider for ETags.
+    /// </summary>
+    public interface ICustomEtagProvider
+    {
+        string CalculateEtagHash(IFileInfo fileInfo);
```

Comment:
Basically, I want to include app version as one of the parameter of this hash function and calculate MD5 hash like this:

```
class StrongEtagProvider : ICustomEtagProvider
    {
        private readonly byte[] appBuildVersionBytes_;

        public StrongEtagProvider(string appBuildVersion)
        {
            appBuildVersionBytes_ = Encoding.UTF8.GetBytes(appBuildVersion);
        }

        public string CalculateEtagHash(IFileInfo fileInfo)
        {
            // MD5_HASH(file-last-modified, file-length, appBuildVersion). This is more reliable than default XOR_HASH(file-last-modified, file-length).
            var buffer = BitConverter.GetBytes(fileInfo.LastModified.ToUniversalTime().Ticks)
                .Concat(BitConverter.GetBytes(fileInfo.Length))
                .Concat(appBuildVersionBytes_)
                .ToArray();
            return GetMd5Hash(new MemoryStream(buffer));
        }

        private static string GetMd5Hash(Stream stream)
        {
            using (var md5 = MD5.Create())
            {
                var hash = md5.ComputeHash(stream);
                return String.Concat(hash.Select(x => x.ToString("X2")));
            }
        }
    }
```
Also, sometimes it may be useful if file name itself has md5/sha1 hash pre-calculated, e.g., WebPack has abilities to generate bundles with MD5 in their names.